test/memory/lifetime/allocation.chpl

```diff
@@ -0,0 +1,81 @@
+use Time;
+
+// Performs multiple trial runs to obtain the average over; we skip the first
+// trial to not only 'warm up the cache' but also give jemalloc the chance to
+// warmup as well, allowing it to request more memory for the heap upfront rather
+// than on-demand.
+config const printTiming : bool = true;
+config const numTrials : int = 3;
+config const allocationsPerTrial : int = 16 * 1024 * 1024;
+
+proc doUnmanagedAllocation() {
+  var timer = new Timer();
+  
+  
+  var times : [1..numTrials] real;
+  for trial in 0..numTrials {
+    {
+      timer.start();
+      var arr : [1..allocationsPerTrial] unmanaged object;
```

Comment:
Just curious -- is it intentional to measure array alloc/init time?

Reply:
I believe so, because there can be additional overhead from initializing the raw object versus the `_shared`/`_owned` wrappers, which gets measured here. As well, it feels uneven, since we _have_ to measure deallocation/deinit of arrays since otherwise the managed types do not get cleaned up, so it makes it more 'even'.